ext/civi_event/Civi/Api4/Action/Participant/Validate.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Api4\Action\Participant;
+
+use Civi\Api4\Generic\ValidateAction;
+use Civi\Api4\Event\ValidateValuesEvent;
+use Civi\API\EntityLookupTrait;
+
+/**
+ * Validate membership parameters before creating/updating Memberships.
+ */
+class Validate extends ValidateAction {
+  use EntityLookupTrait;
+
+  public $context = [];
+
+  protected function onValidateValues(ValidateValuesEvent $e) {
+    foreach ($e->records as $recordKey => $record) {
+      parent::onValidateValues($e);
+      if (!empty($e->event_id) && !$this->isDefined('Event' . $e->event_id)) {
+        $this->define('Event', 'Event' . $e->event_id, ['id' => $e->event_id]);
+      }
+      if (!empty($record['id'])) {
+        $this->validateCreate($record, $recordKey);
+      }
+      else {
+        $this->validateUpdate($record, $recordKey);
+      }
```

Comment:
Per comments on the other validate PRs, I think these would be better added in as event subscribers so we're not stuck with hardcoded validation that can't easily be changed.